implementations/elixir/ockam/ockam/lib/ockam/transport/tcp/listener.ex

```diff
@@ -163,19 +167,29 @@ if Code.ensure_loaded?(:ranch) do
 
     @impl true
     def handle_info({:tcp, socket, data}, %{socket: socket, address: address} = state) do
+      {function_name, _} = __ENV__.function
+
       with {:ok, decoded} <- Ockam.Wire.decode(@wire_encoder_decoder, data),
            {:ok, decoded} <- set_return_route(decoded, address) do
         send_to_router(decoded)
+        Telemetry.emit_event(function_name, metadata: %{name: "decoded_data"})
       else
-        {:error, %Ockam.Wire.DecodeError{} = e} -> raise e
-        e -> raise e
+        {:error, %Ockam.Wire.DecodeError{} = e} ->
+          raise e
```

Comment:
I'm pretty sure this raise needs to be moved below the call to `Telemetry.emit_exception_event/3` so that the event will actually be emitted.

Reply:
You are of course right. Telemetry.emit_exception_event will not be called because the exception will be raised earlier.
How about now?

In match for {{:error, %Ockam.Wire.DecodeError{} = e}} - decode error) Telemetry event will be triggered at first and then the exception will be raised afterwards.